datafusion/datasource/src/file_scan_config.rs

```diff
@@ -785,19 +790,24 @@ impl FileScanConfig {
     /// Project the schema, constraints, and the statistics on the given column indices
     pub fn project(&self) -> (SchemaRef, Constraints, Statistics, Vec<LexOrdering>) {
         if self.projection.is_none() && self.table_partition_cols.is_empty() {
+            let output_ordering: Vec<LexOrdering> =
```

Comment:
I'm not convinced that this is correct, with the same argument as for the display code. I think the return type here also needs to be `Vec<Option<...>>`.